libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -728,6 +728,53 @@ void RCOutput::set_output_mode(uint16_t mask, enum output_mode mode)
 #endif
 }
 
+/*
+ * get output mode banner to inform user of how outputs are configured
+ */
+bool RCOutput::get_output_mode_banner(char banner_msg[], uint8_t banner_msg_len) const
+{
+    // create array of each channel's mode
+    output_mode ch_mode[NUM_GROUPS * ARRAY_SIZE(pwm_group::chan)] = {MODE_PWM_NONE};
+    bool have_nonzero_modes = false;
+    for (uint8_t i = 0; i < NUM_GROUPS; i++ ) {
+        pwm_group &group = pwm_group_list[i];
+        if (group.current_mode != MODE_PWM_NONE) {
+            for (uint8_t j = 0; j < ARRAY_SIZE(group.chan); j++) {
+                if (group.chan[j] < ARRAY_SIZE(ch_mode)) {
+                    ch_mode[group.chan[j]] = group.current_mode;
+                    have_nonzero_modes = true;
+                }
+            }
+        }
+    }
+
+    // handle simple case
+    if (!have_nonzero_modes) {
+        hal.util->snprintf(banner_msg, banner_msg_len, "RCOut: None");
```

Comment:
Not sure this is informative enough.  How about "RCOut: no RC outputs defined"?

Reply:
hmm.. not sure.. the new suggestion just leads to "RC" and "Out" being repeated.  "defined" is the only extra bit of info and I'm not sure that adds to much.  Sorry, I think I might leave it as-is.